scripts/index.js

```diff
@@ -0,0 +1,117 @@
+const state = {
+    temp: 82,
+    sky: "Sunny",
+    city: "Emerald City"
+};
```

Comment:
Grouping your state values together like this can be a helpful way to avoid polluting the global namespace in Javascript. The [Immediately Invoked Function Expression](https://developer.mozilla.org/en-US/docs/Glossary/IIFE) pattern can also help us with that if we wrap our entire function with

```js
(function(){
    // ALL our JS code in this file here
})();
```

Since any variables and functions will effectively be declared local to the wrapping function, which itself has no name. So this construct can give us an area in which to create variables and functions without concern of running into other names.

This would effectively allow us to look up our page elements only once, and have them accessible throughout any helper classes (this would look very similar to React with Hooks). Alternatively, we could write a JS class that does those lookups and stores them for later use during the constructor or some other init method.

With React apps, we will see this approach isn't necessary, since React using JS modules to namespace its variables and functions.